shared/common-adapters/user-proofs.shared.js

```diff
@@ -0,0 +1,81 @@
+/* @flow */
+
+import {globalColors} from '../styles/style-guide'
+import {normal as proofNormal, checking as proofChecking, revoked as proofRevoked, error as proofError, warning as proofWarning} from '../constants/tracker'
+import {metaNew, metaUpgraded, metaUnreachable, metaPending, metaDeleted, metaIgnored} from '../constants/tracker'
+
+import type {Props as IconProps} from '../common-adapters/icon'
+import type {Proof} from './user-proofs'
+
+export function metaColor (proof: Proof): string {
+  let color = globalColors.blue
+  switch (proof.meta) {
+    case metaNew: color = globalColors.blue; break
+    case metaUpgraded: color = globalColors.blue; break
+    case metaUnreachable: color = globalColors.red; break
+    case metaPending: color = globalColors.black_40; break
+    case metaDeleted: color = globalColors.red; break
+    case metaIgnored: color = globalColors.green; break
+  }
+  return color
+}
+
+export function proofColor (proof: Proof): string {
+  let color = globalColors.blue
+  switch (proof.state) {
+    case proofNormal: {
+      color = proof.isTracked ? globalColors.green2 : globalColors.blue
+      break
+    }
+    case proofChecking:
+      color = globalColors.black_20
+      break
+    case proofRevoked:
+    case proofWarning:
+    case proofError:
+      color = globalColors.red
+      break
+  }
+  if (proof.state === proofChecking) color = globalColors.black_20
+  return color
+}
+
+export function iconNameForProof (proof: Proof): IconProps.type {
+  return {
+    'twitter': 'fa-twitter',
+    'github': 'fa-github',
+    'reddit': 'fa-reddit',
+    'pgp': 'fa-key',
+    'coinbase': 'fa-kb-iconfont-coinbase',
+    'hackernews': 'fa-hacker-news',
+    'rooter': 'fa-shopping-basket',
+    'http': 'fa-globe',
+    'https': 'fa-globe',
+    'dns': 'fa-globe',
+  }[proof.type]
+}
+
+export function proofStatusIcon (proof: Proof): ?IconProps.type {
+  switch (proof.state) {
+    case proofChecking:
+      return 'fa-kb-iconfont-proof-pending'
+
+    case proofNormal:
+      return proof.isTracked ? 'fa-kb-iconfont-proof-followed' : 'fa-kb-iconfont-proof-new'
+
+    case proofWarning:
+    case proofError:
+    case proofRevoked:
+      return 'fa-kb-iconfont-proof-broken'
+
+    default:
+      return null
+  }
+}
+
+export function proofNameStyle (proof: Proof) {
+  return {
+    color: proofColor(proof),
+    ...(proof.meta === metaDeleted ? {textDecorationLine: 'line-through'} : {}),
```

Comment:
yeah you have to be updated, our rn version previously didn't have it

On Thu, Jun 23, 2016 at 1:17 PM, Alex Wendland <notifications@github.com>
wrote:

> In shared/common-adapters/user-proofs.shared.js
> <https://github.com/keybase/client/pull/3172#discussion_r68275896>:
>
> > +      return proof.isTracked ? 'fa-kb-iconfont-proof-followed' : 'fa-kb-iconfont-proof-new'
> > +
> > +    case proofWarning:
> > +    case proofError:
> > +    case proofRevoked:
> > +      return 'fa-kb-iconfont-proof-broken'
> > +
> > +    default:
> > +      return null
> > +  }
> > +}
> > +
> > +export function proofNameStyle (proof: Proof) {
> > +  return {
> > +    color: proofColor(proof),
> > +    ...(proof.meta === metaDeleted ? {textDecorationLine: 'line-through'} : {}),
>
> I'll need to test further to see why it's only showing up on iOS not
> Android then. Maybe I'm not working off of master with the latest RN.
>
> —
> You are receiving this because you are on a team that was mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/keybase/client/pull/3172/files/789b5b5761cc042be78274e468669bab66139696#r68275896>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AAfhU58xymJ2eq_b-swTAa8QXfYCTTOIks5qOr-jgaJpZM4I8U7j>
> .
>